pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -686,6 +686,13 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontaine
 		klog.V(4).Infof("Creating sandbox for pod %q", format.Pod(pod))
 		createSandboxResult := kubecontainer.NewSyncResult(kubecontainer.CreatePodSandbox, format.Pod(pod))
 		result.AddSyncResult(createSandboxResult)
+		backOffKey := fmt.Sprintf("%s_%s_%s", pod.Name, pod.Namespace, string(pod.UID))
```

Comment:
Thoughts on whether we could contain this backoff behavior within the `kubeGenericRuntimeManager.createPodSandbox` method?

I feel like that would have the benefit of decreasing the amount of code in `kubeGenericRuntimeManager.SyncPod` (which is already quite long).

Reply:
> Thoughts on whether we could contain this backoff behavior within the `kubeGenericRuntimeManager.createPodSandbox` method?
Agree with this.  how about add backoff in `kubeGenericRuntimeManager.createPodSandbox` for func `RunPodSandbox` ?
```
--- a/pkg/kubelet/kuberuntime/kuberuntime_sandbox.go
+++ b/pkg/kubelet/kuberuntime/kuberuntime_sandbox.go
@@ -58,11 +58,18 @@ func (m *kubeGenericRuntimeManager) createPodSandbox(pod *v1.Pod, attempt uint32
                        return "", message, err
                }
        }
-
+       backOffKey := fmt.Sprintf("%s_%s_%s", pod.Name, pod.Namespace, string(pod.UID))
+       if backOff.IsInBackOffSinceUpdate(backOffKey, backOff.Clock.Now()) {
+               msg := fmt.Sprintf("Back-off create sandbox for pod %q", format.Pod(pod))
+               klog.Infof(msg)
+               createSandboxResult.Fail(kubecontainer.ErrCreatePodSandbox, msg)
+               return
+       }
        podSandBoxID, err := m.runtimeService.RunPodSandbox(podSandboxConfig, runtimeHandler)
        if err != nil {
                message := fmt.Sprintf("CreatePodSandbox for pod %q failed: %v", format.Pod(pod), err)
                klog.Error(message)
+               backOff.Next(backOffKey, backOff.Clock.Now())
                return "", message, err
        }
```


> 
> I feel like that would have the benefit of decreasing the amount of code in `kubeGenericRuntimeManager.SyncPod` (which is already quite long).